cluster-autoscaler/cloudprovider/azure/azure_cache.go

```diff
@@ -20,97 +20,244 @@ import (
 	"regexp"
 	"strings"
 	"sync"
+	"time"
 
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-12-01/compute"
+	"github.com/Azure/go-autorest/autorest/to"
 	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
-	klog "k8s.io/klog/v2"
+
+	"k8s.io/klog/v2"
 )
 
 var (
 	virtualMachineRE = regexp.MustCompile(`^azure://(?:.*)/providers/Microsoft.Compute/virtualMachines/(.+)$`)
 )
 
-type asgCache struct {
-	registeredAsgs     []cloudprovider.NodeGroup
-	instanceToAsg      map[azureRef]cloudprovider.NodeGroup
-	notInRegisteredAsg map[azureRef]bool
-	mutex              sync.Mutex
-	interrupt          chan struct{}
+// azureCache is used for caching cluster resources state.
+//
+// It is needed to:
+// - keep track of node groups (AKS, VM and VMSS types) in the cluster,
+// - keep track of instances and which node group they belong to,
+// - limit repetitive Azure API calls.
+type azureCache struct {
+	mutex           sync.Mutex
+	interrupt       chan struct{}
```

Comment:
what is this channel used for?

Reply:
I'm not entirely sure, it was there before https://github.com/kubernetes/autoscaler/pull/3717/files#diff-63481ca096e322d8f48e57a4b21089a30481cc3798e3dc123a1b4ba0938ceb1dL37